File: 1480. Running Sum of 1d Array/runningSum.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var runningSum = function(nums) {
    for (let i=1;i<nums.length;i++){
        nums[i] += nums[i-1]
    }
    return nums
};

// Runtime 81 ms / Memory 42.5 MB
// debug: https://jsfiddle.net/

File: 1480. Running Sum of 1d Array/runningSum.cpp
class Solution {
public:
    vector<int> runningSum(vector<int>& nums) {
        // start from inedx 1
        for (int i=1; i<nums.size(); i++) {
            nums[i] += nums[i-1]; //accumulate
        }
        return nums;
    }
};